src/backend/distributed/operations/shard_split.c

```diff
@@ -431,22 +465,47 @@ SplitShard(SplitMode splitMode,
 			lappend(workersForPlacementList, (void *) workerNode);
 	}
 
+	List *sourceColocatedShardIntervalList = NIL;
+	if (splitOperation == CREATE_DISTRIBUTED_TABLE)
+	{
+		/*
+		 * there is no colocation entries yet for local table, so we should
+		 * check if table has any partition and add them to same colocation
+		 * shard interval group
+		 */
+		sourceColocatedShardIntervalList = ListShardsUnderParentRelation(
```

Comment:
I'm wondering whether we should move the responsibility of calling sourceColocatedShardIntervalList to the caller. I could imagine a scenario where we want to expand create_distributed_table_concurrently to do multiple tables at once (e.g. cascade via foreign keys, or provide a list).

It's also weird for a function to do different things depending on who calls it.